deeplearning4j-ui-parent/deeplearning4j-play/src/main/java/org/deeplearning4j/ui/play/PlayUIServer.java

```diff
@@ -222,7 +230,20 @@ public void runMain(String[] args) {
         uiEventRoutingThread = new Thread(new StatsEventRouterRunnable());
         uiEventRoutingThread.setDaemon(true);
         uiEventRoutingThread.start();
-        if (enableRemote)
+        if (enableFileStorage && customStatsFile != null) {
+            enableRemoteListener(new FileStatsStorage(new File(customStatsFile)), true);
+        }
+        else if(enableFileStorage) {
+            try {
+                File tempStatsFile = File.createTempFile("dl4j", "UIstats");
+                tempStatsFile.deleteOnExit();
+                enableRemoteListener(new FileStatsStorage(tempStatsFile), true);
+            } catch(Exception e) {
+                System.out.println("Failed to create temporary file for stats storage");
+                System.exit(0);
+            }
+        }
+        else if(enableRemote)
             enableRemoteListener();
```

Comment:
On second thought - let's merge enableRemote and enableFileStorage - i.e., default to FileStorage only (temp dir unless path is set). File has better performance and less memory use...